core-build/gulp-core-build-typescript/src/miniTasks/TscCmdTask.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as Gulp from 'gulp';
+import * as path from 'path';
+import {
+  JsonFile,
+  IPackageJson
+} from '@microsoft/node-core-library';
+import * as fsx from 'fs-extra';
+import * as glob from 'glob';
+
+import { BaseCmdTask } from './BaseCmdTask';
+
+/**
+ * @public
+ */
+export interface ITscCmdTaskConfig {
+  /**
+   * The path to the typescript compiler package if the task should override the version of the compiler.
+   */
+  typescriptCompilerPackagePath?: string;
+
+  /**
+   * Optional list of custom args to pass to "tsc"
+   */
+  customArgs?: string[];
+
+  /**
+   * Glob matches for files to be passed through the build.
+   */
+  staticMatch?: string[];
+
+  /**
+   * The directory in which the typescript compiler should be invoked.
+   */
+  buildDirectory?: string;
+}
+
+/**
+ * @alpha
+ */
+export class TscCmdTask extends BaseCmdTask<ITscCmdTaskConfig> {
+  constructor() {
+    super(
+      'tsc',
+      {
+        staticMatch: [
+          'src/**/*.js',
+          'src/**/*.json',
+          'src/**/*.jsx'
+        ]
+      }
+    );
+  }
+
+  public loadSchema(): Object {
+    return JsonFile.load(path.resolve(__dirname, '..', 'schemas', 'tsc-cmd.schema.json'));
+  }
+
+  public executeTask(gulp: typeof Gulp, completeCallback: (error?: string) => void): Promise<void> | undefined {
+    // require.resolve('typescript') resolves to <package>/lib/typescript.js, and we want the package path
+    let compilerPackagePath: string = path.resolve(require.resolve('typescript'), '..', '..');
```

Comment:
>'), '..', '..'); [](start = 78, length = 16)

this is hardcoded, since you load the package.json later, we could future proof by crawling upwards looking for package.json